### ISSUE TITLE
raises focus onto window behind menu bar dropdown of little snitch

### ISSUE THREAD

**Issue by Lexvox** [sbmpost/AutoRaise#205]
Problem:
when clicking on a menu bar item of Little Snitch focus will get raised to the window beneath the dropdown.
So far this only hapens with Little Snitch, all other apps work as expected, not raising focus to a window beneath a menu bar dropdown.

Expected:
auto raise doesn't focus on a window behind a menu bar dropdown on all aps consitently

Reproduce:
Open Little Snitch Rules in windowed mode, place it over another app eg. Finder, so that the window of the other app is directly beneath the menu bar.
Focus Little Snitch, go to menu bar with cursor and open drop down.
Hover over a menu dropdown item.
Focus will shift to window beneath menu dropdown.

Window raise is disabled.
Screenrecord attached.

https://github.com/sbmpost/AutoRaise/assets/53237831/9322d2d7-cbbf-4fe0-baef-a7a45e64466d

**Comment by sbmpost** [comment]
@Lexvox 

See issue #204.

**Comment by sbmpost** [comment]
@Lexvox 

Can I close this issue? The solution presented in #204 should get you going.

**Comment by Lexvox** [comment]
I tried the solution in #204 and it doesn't change the behavior. Only change is that it breaks the settings ui somewhat. 
<img width="507" alt="Screenshot 2024-04-23 at 16 35 10" src="https://github.com/sbmpost/AutoRaise/assets/53237831/689a1ab5-6c73-49c4-9297-ddee2bf429ad">

Restarted, no change. 

I did not manage to use the command 
`./AutoRaise -focusDelay 1 -stayFocusedBundleIds "at.obdev.littlesnitch,at.obdev.littlesnitch.agent,at.obdev.littlesnitch.networkmonitor"`

that you shared in #204 what dir do I run this in?

**Comment by sbmpost** [comment]
The stayFocusedBundleIds is a comma separated list without spaces in the UI as well.

So if you copy paste this:

    at.obdev.littlesnitch,at.obdev.littlesnitch.agent,
at.obdev.littlesnitch.networkmonitor

And confirm with the enter key in the UI it should work. Hope that helps.

**Comment by Lexvox** [comment]
I did exactly that and did confirm with the enter key (and settings persisted the reboot)

**Comment by sbmpost** [comment]
@Lexvox 

And the problem in the video persists I assume? The focus being stolen unexpectedly doesn't occur on my system with the settings I provided. So let's try to figure out why it still breaks on your system. One thing I can think of, is that you may be running two AutoRaise instances at the same time somehow. You can check this with the "Activity Monitor" app. If you (force) stop all AutoRaise instances from there, and then:

    1. Download the master branch.
    2. Unpack as per the instructions in the readme
    3. Make the project as per the instructions in the readme
    4. Start AutoRaise from the commandline with the stayFocusedBundleIds settings

Then hopefully we can see what is going on. I realize doing steps 1 - 4 might be a bit of a hassle, but probably the best place to start from in order to crush this issue ;-)

**Comment by Lexvox** [comment]
wierd.

does this mean theres two instances running bc if I understood correctly one of them is the actual app and the other one is simply the gui.
<img width="899" alt="Screenshot 2024-04-28 at 14 40 40" src="https://github.com/sbmpost/AutoRaise/assets/53237831/6968b72a-436f-40cc-873d-0bad1dd87490">

I followed ur steps and with it running from my terminal it worked as expected.

reinstalled the app and after adding Little Snitch to stay focused everything works as expected. 

also the ui glitch I showed in the last screenshot doesn't appear again.

I guess we can close this then. Guess there was something bugged abt my previous install / bug in front of the screen. 🙄

Thanks for your time and thanks for this very crucial app. Still boggles my mind this isn't just an option in macos.
